beacon-chain/rpc/eth/validator/validator.go

```diff
@@ -175,45 +186,55 @@ func (vs *Server) GetSyncCommitteeDuties(ctx context.Context, req *ethpbv2.SyncC
 		return nil, status.Error(codes.Unavailable, "Syncing to latest head, not ready to respond")
 	}
 
-	slot, err := slots.EpochStart(req.Epoch)
+	currentSlot := slots.CurrentSlot(uint64(vs.TimeFetcher.GenesisTime().Unix()))
+	currentEpoch := slots.ToEpoch(currentSlot)
+	lastValidEpoch := syncCommitteeDutiesLastValidEpoch(currentEpoch)
+	if req.Epoch > lastValidEpoch {
+		return nil, status.Errorf(codes.InvalidArgument, "Epoch is too far in the future. Maximum valid epoch is %v.", lastValidEpoch)
+	}
+
+	var stateEpoch types.Epoch
+	// Integer division and multiplication do not cancel out.
+	currentSyncCommitteeFirstEpoch := (currentEpoch / params.BeaconConfig().EpochsPerSyncCommitteePeriod) * params.BeaconConfig().EpochsPerSyncCommitteePeriod
+	if req.Epoch >= currentSyncCommitteeFirstEpoch {
+		stateEpoch = currentSyncCommitteeFirstEpoch
+	} else {
+		stateEpoch = req.Epoch
```

Comment:
this is not correct, `stateEpoch` should be `slots.SyncCommitteePeriodStartEpoch(req.epoch)`

Reply:
Does it make a difference? Won't duties be the same throughout the whole period? I will change this to what you recommend either way.

Reply:
it wont make a difference here as i read through the function, luckily we are just getting the committee from the state, we are not sampling the committee. To sample the committee, we should always use the epoch boundary state as described here https://github.com/ethereum/consensus-specs/pull/2394